impeller/renderer/backend/metal/compute_pass_mtl.mm

```diff
@@ -0,0 +1,251 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/renderer/backend/metal/compute_pass_mtl.h"
+
+#include <Metal/Metal.h>
+#include <memory>
+#include <variant>
+
+#include "flutter/fml/closure.h"
+#include "flutter/fml/logging.h"
+#include "flutter/fml/trace_event.h"
+#include "impeller/base/backend_cast.h"
+#include "impeller/renderer/backend/metal/compute_pipeline_mtl.h"
+#include "impeller/renderer/backend/metal/device_buffer_mtl.h"
+#include "impeller/renderer/backend/metal/formats_mtl.h"
+#include "impeller/renderer/backend/metal/sampler_mtl.h"
+#include "impeller/renderer/backend/metal/texture_mtl.h"
+#include "impeller/renderer/compute_command.h"
+#include "impeller/renderer/formats.h"
+#include "impeller/renderer/host_buffer.h"
+#include "impeller/renderer/shader_types.h"
+
+namespace impeller {
+
+ComputePassMTL::ComputePassMTL(std::weak_ptr<const Context> context,
+                               id<MTLCommandBuffer> buffer)
+    : ComputePass(std::move(context)), buffer_(buffer) {
+  if (!buffer_) {
+    return;
+  }
+  is_valid_ = true;
+}
+
+ComputePassMTL::~ComputePassMTL() = default;
+
+bool ComputePassMTL::IsValid() const {
+  return is_valid_;
+}
+
+void ComputePassMTL::OnSetLabel(std::string label) {
+  if (label.empty()) {
+    return;
+  }
+  label_ = std::move(label);
+}
+
+bool ComputePassMTL::OnEncodeCommands(const Context& context) const {
+  TRACE_EVENT0("impeller", "ComputePassMTL::EncodeCommands");
+  if (!IsValid()) {
+    return false;
+  }
+
+  // TODO(dnfield): Support non-serial dispatch type on higher iOS versions.
+  auto compute_command_encoder = [buffer_ computeCommandEncoder];
```

Comment:
We should be able to support this use cases using just creating multiple compute passes. This, instead of relying on the Metal API feature. I think we can get rid of this TODO.

Reply:
My assumption is that Vulkan would have similar API for collecting compute information, using whatever necessary barriers, whether for use in a subsequent vertex shader or back on the host.